installTerraInVm.sh

```diff
@@ -1,53 +1,54 @@
-# first make sure all terraform dependencies are installed
+#!/bin/bash
+
 # Check if gnupg is installed
-if which gpg > /dev/null 2>&1; then
-  echo 'gnupg and software-properties-common are already installed'
+if gpg --version; then
+    echo 'gnupg is already installed'
 else
-  echo 'Installing gnupg and software-properties-common...'
-  sudo apt-get update && sudo apt-get install -y gnupg software-properties-common
+    sudo apt-get update
+    echo 'Installing gnupg and software-properties-common...'
+    sudo apt-get install -y gnupg software-properties-common
 fi
 
-# Check if the HashiCorp GPG key is present
-if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
-  echo 'HashiCorp GPG key is already present.'
+# Check for apt-cache status
+if apt-cache search gnupg && apt-cache search software-properties-common; then
```

Comment:
`apt-cache show gnupg` would be good enough. Be careful putting logical AND in your tests (desired state)